From d7630194b29919fd5c23709692576de365b256a6 Mon Sep 17 00:00:00 2001
From: Xu Liang <liang.xu@intel.com>
Date: Mon, 15 Jun 2020 14:19:24 +0800
Subject: [PATCH] DRVLIB_SW-3690: GSW API - fix input boundary check of
 GSW_CapGet and add entry GSW_CAP_TYPE_COMMON_PCE_RULES

---
 drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c b/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c
index 625b8ed57e8b..01676f9107cc 100644
--- a/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c
+++ b/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c
@@ -188,6 +188,7 @@ static gsw_capdesc_t capdes[] = {
 	{ GSW_CAP_TYPE_RT_RTP, "Number of RTP Entries:"},
 	{ GSW_CAP_TYPE_CTP, "Number of CTP ports"},
 	{ GSW_CAP_TYPE_BRIDGE_PORT, "Number of bridge ports"},
+	{ GSW_CAP_TYPE_COMMON_PCE_RULES, "Number of global PCE rules"},
 	{ GSW_CAP_TYPE_LAST, "Last Capability Index"}
 };
 
@@ -11940,7 +11941,8 @@ GSW_return_t GSW_CapGet(void *cdev, GSW_cap_t *parm)
 	spin_lock_bh(&gswdev->lock_misc);
 #endif
 
-	if (parm->nCapType >= GSW_CAP_TYPE_LAST) {
+	if (parm->nCapType >= GSW_CAP_TYPE_LAST
+	    || parm->nCapType >= ARRAY_SIZE(capdes)) {
 		ret = GSW_statusErr;
 		goto UNLOCK_AND_RETURN;
 	} else {
